src/state/lottery/fetchUnclaimedUserRewards.ts

```diff
@@ -0,0 +1,190 @@
+  console.log('winning tickets: ', ticketsWithRewards)
+  return { ticketsWithRewards, cakeTotal }
+}
+
+const getRewardBracket = (ticketNumber: string, finalNumber: string): number => {
+  // Winning numbers are evaluated right-to-left in the smart contract, so we reverse their order for validation here:
+  // i.e. '1123456' should be evaluated as '6543211'
+  const ticketNumAsArray = ticketNumber.split('').reverse()
+  const winningNumsAsArray = finalNumber.split('').reverse()
+  const matchingNumbers = []
+
+  // The number at index 6 in all tickets is 1 and will always match, so finish at index 5
+  for (let index = 0; index < winningNumsAsArray.length - 1; index++) {
+    if (ticketNumAsArray[index] !== winningNumsAsArray[index]) {
+      break
+    }
+    matchingNumbers.push(ticketNumAsArray[index])
+  }
+
+  // Reward brackets refer to indexes, 0 = 1 match, 5 = 6 matches. Deduct 1 from matchingNumbers' length to get the reward bracket
+  const rewardBracket = matchingNumbers.length - 1
+  return rewardBracket
+}
+
+const getWinningTickets = async (roundDataAndUserTickets: RoundDataAndUserTickets): Promise<LotteryTicketClaimData> => {
+  const { roundId, userTickets, finalNumber } = roundDataAndUserTickets
+
+  const ticketsWithRewardBrackets = userTickets.map((ticket) => {
+    return {
+      roundId,
+      id: ticket.id,
+      number: ticket.number,
+      status: ticket.status,
+      rewardBracket: getRewardBracket(ticket.number, finalNumber),
+    }
+  })
+  // TODO: Remove log. To help in testing when verifying winning tickets.
+  console.log(`all tickets round #${roundId}`, ticketsWithRewardBrackets)
+
+  // A rewardBracket of -1 means no matches. 0 and above means there has been a match
+  // If ticket.status is true, the ticket has already been claimed
+  const winningTickets = ticketsWithRewardBrackets.filter((ticket) => {
+    return ticket.rewardBracket >= 0 && !ticket.status
+  })
+
+  if (winningTickets.length > 0) {
+    const { ticketsWithRewards, cakeTotal } = await fetchCakeRewardsForTickets(winningTickets)
+    return { ticketsWithRewards, cakeTotal, roundId }
+  }
+  return null
+}
+
+const getWinningNumbersForRound = (targetRoundId: string, lotteriesData: LotteryRoundGraphEntity[]) => {
+  const targetRound = lotteriesData.find((pastLottery) => pastLottery.id === targetRoundId)
+  return targetRound?.finalNumber
+}
+
+const fetchUserTicketsForMultipleRounds = async (roundsToCheck: UserRound[], account: string) => {
+  // Build calls with data to help with merging multicall responses
+  const callsWithRoundData = roundsToCheck.map((round) => {
+    const totalTickets = parseInt(round.totalTickets, 10)
+    const calls = getViewUserTicketInfoCalls(totalTickets, account, round.lotteryId)
+    return { calls, lotteryId: round.lotteryId, count: calls.length }
+  })
+
+  // Batch all calls across all rounds
+  const multicalls = [].concat(...callsWithRoundData.map((callWithRoundData) => callWithRoundData.calls))
```

Comment:
So `callsWithRoundData` is an array of different objects, I map over them to extract just the calls... but I don't want these returned in an array of arrays (I want one array with all the calls - so they can be batched as a single call). 

Spreading the return of the map into `[].concat(...` was one way to achieve this. Maybe there's a better way?

Reply:
could be `callsWithRoundData.flatMap((callWithRoundData) => callWithRoundData.calls)`
That would be transpiled into something like you have though cause .flatMap is only available since ES2019 and project target is ES5.
Could be changed but not critical

Reply:
Added a note to the refactors trello, as you say - not sure it's critical or even necessary.